File: result/translate/transcoder/python2java/java/SUBARRAYS_DISTINCT_ELEMENTS.java
public static int f_gold ( int [ ] arr , int n ) {
  ArrayList < Integer > s = new ArrayList < Integer > ( ) ;
  int j = 0 ;
  int ans = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    while ( ( j < n ) && ( arr [ j ] != null ) ) {
      s . add ( arr [ j ] ) ;
      j ++ ;
    }
    ans += ( ( j - i ) * ( j - i + 1 ) ) / 2 ;
    s . remove ( arr [ i ] ) ;
  }
  return ans ;
}

File: result/repair/starchat/java2cpp/c++_repair/SUBARRAYS_DISTINCT_ELEMENTS.cpp
int f_gold(vector<int> arr, int n) {
int[] s;
    int j = 0;
    int ans = 0;
    for (int i = 0; i < n; i++) {
        while (j < n &&!s.count(arr[j])) {
            s.insert(arr[j]);
            j++;
        }
        ans += ((j - i) * (j - i + 1)) / 2;
        s.erase(arr[i]);
    }
    return ans;
}
